src/framework/auth/strategies/oauth2/oauth2-strategy.ts

```diff
@@ -154,17 +154,22 @@ export class NbOAuth2AuthStrategy extends NbAuthStrategy {
     super();
   }
 
-  authenticate(): Observable<NbAuthResult> {
-    return this.isRedirectResult()
-      .pipe(
-        switchMap((result: boolean) => {
-          if (!result) {
-            this.authorizeRedirect();
-            return observableOf(new NbAuthResult(true));
-          }
-          return this.getAuthorizationResult();
-        }),
-      );
+  authenticate(data?: any): Observable<NbAuthResult> {
+
+    if (this.getOption('password') && data) {
```

Comment:
there is no need to check on the `data` I believe. Probably we better check  it like this:
```
if (this.getOption('token.grantType') === NbOAuth2GrantType.PASSWORD) {
} else {
 //
}
```

Reply:
Test:wp failed as i was not checking data. 
Your proposition is great, going to try.